cmd/dep/init.go

```diff
@@ -115,44 +122,59 @@ func (cmd *initCommand) Run(ctx *dep.Ctx, args []string) error {
 	sm.UseDefaultSignalHandling()
 	defer sm.Release()
 
-	ctx.Loggers.Err.Println("Searching GOPATH for projects...")
-	pd, err := getProjectData(ctx, pkgT, cpr, sm)
-	if err != nil {
-		return err
-	}
+	// Create empty manifest, lock and projectData, and fill them according to
+	// init operation modes. If gopath flag is set, fetch projectData by searching
+	// through GOPATH, else operate in network mode, solve all the dependencies
+	// over network.
 	m := &dep.Manifest{
-		Dependencies: pd.constraints,
+		Dependencies: make(gps.ProjectConstraints),
 	}
+	l := &dep.Lock{}
+	var pd projectData
 
-	// Make an initial lock from what knowledge we've collected about the
-	// versions on disk
-	l := &dep.Lock{
-		P: make([]gps.LockedProject, 0, len(pd.ondisk)),
-	}
+	if cmd.gopath {
```

Comment:
Here's where the various bits are in gopath_scanner:

* Scan to generate project data -> gopathScanner.scanGopathForDependencies
* Populate a manifest using the project data -> gopathScanner.InitializeRootManifestAndLock
* Combine the results of the importer with what we found in GOPATH, printing feedback for what was used (things from the importers win) -> gopathScanner.overlay
* Tweak the lock based on the results of solving -> gopathScanner.FinalizeRootManifestAndLock